server/src/main/java/org/opensearch/common/blobstore/stream/read/ReadContext.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.blobstore.stream.read;
+
+import java.io.InputStream;
+import java.util.List;
+
+/**
+ * ReadContext is used to encapsulate all data needed by <code>VerifyingMultiStreamBlobContainer#asyncBlobDownload</code>
+ *
+ * @opensearch.experimental
+ */
+public class ReadContext {
+    private final List<InputStream> blobInputStreams;
```

Comment:
can we make this a supplier of streams just like `WriteContext` ? This loads the streams lazily on demand .